src/routes/community/community.controller.spec.ts

```diff
@@ -279,6 +283,155 @@ describe('Community (Unit)', () => {
     });
   });
 
+  describe('GET /campaigns/:resourceId/activity', () => {
+    it('should get the campaign activity by campaign ID', async () => {
+      const campaign = campaignBuilder().build();
+      const campaignActivity = campaignActivityBuilder().build();
+      const campaignActivityPage = pageBuilder()
+        .with('results', [campaignActivity])
+        .with('count', 1)
+        .with('previous', null)
+        .with('next', null)
+        .build();
+      networkService.get.mockImplementation(({ url }) => {
+        switch (url) {
+          case `${lockingBaseUri}/api/v1/campaigns/${campaign.resourceId}/activities`:
+            return Promise.resolve({ data: campaignActivityPage, status: 200 });
+          default:
+            return Promise.reject(`No matching rule for url: ${url}`);
+        }
+      });
+
+      await request(app.getHttpServer())
+        .get(`/v1/community/campaigns/${campaign.resourceId}/activities`)
+        .expect(200)
+        .expect({
+          count: 1,
+          next: null,
+          previous: null,
+          results: [campaignActivityToJson(campaignActivity)],
+        });
+    });
+
+    it('should get the campaign activity by campaign ID and holder', async () => {
+      const campaign = campaignBuilder().build();
+      const holder = getAddress(faker.finance.ethereumAddress());
+      const campaignActivity = campaignActivityBuilder()
+        .with('holder', holder)
+        .build();
+      const campaignActivityPage = pageBuilder()
+        .with('results', [campaignActivity])
+        .with('count', 1)
+        .with('previous', null)
+        .with('next', null)
+        .build();
+      networkService.get.mockImplementation(({ url }) => {
+        switch (url) {
+          case `${lockingBaseUri}/api/v1/campaigns/${campaign.resourceId}/activities`:
+            return Promise.resolve({ data: campaignActivityPage, status: 200 });
+          default:
+            return Promise.reject(`No matching rule for url: ${url}`);
+        }
+      });
+
+      await request(app.getHttpServer())
+        .get(
+          `/v1/community/campaigns/${campaign.resourceId}/activities?holder=${holder}`,
+        )
+        .expect(200)
+        .expect({
+          count: 1,
+          next: null,
+          previous: null,
+          results: [campaignActivityToJson(campaignActivity)],
+        });
```

Comment:
Nit: could we include a test verifying the `networkService` GET call included the `holder`, `limit` and `offset` parameter?

Reply:
Added in 724dc1318d1bd6cb4b5ca3fc141930b0f4edbbe5.